File: algoritmos/2_super_classe_animal/animais/pato.dart
import '../interfaces/acao_animal_interface.dart';
import '../tipos/ave.dart';

class Pato extends Ave implements AcaoAnimalInterface {
  String cor;

  Pato(this.cor, String sexo, int idade, double peso, int quantidadeDePatas)
      : super(sexo, idade, peso, quantidadeDePatas);

  @override
  void voar() {
    print('Os patos voam!');
  }
}

File: algoritmos/2/challenge-opp/Entities/Pato.cs
using challenge_opp.Interfaces;

namespace challenge_opp.Entities;
public sealed class Pato : Ave, IVoo
{
    public Pato(double peso, int quantidadeDePatas) : base(peso, quantidadeDePatas)
    {
    }

    public void Voar() =>
        Console.WriteLine("Pato voa");
}
